src/averageAges.js

```diff
@@ -37,7 +41,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = (!withChildren)
+    ? people.filter(gender => gender.sex === 'f')
+    : people.filter(mother => mother.sex === 'f'
+    && (people.find(person => mother.name === person.mother)));
```

Comment:
[checklist points 2, 3](https://github.com/mate-academy/js_average-ages/blob/master/checklist.md)

It would be more clearly if you would create a variable `isMother` and method `some` would be better use for this purpose then `find`.